File: src/main/java/dev/isxander/evergreenhud/elements/impl/ElementPitch.java
/*
 * Copyright (C) isXander [2019 - 2021]
 * This program comes with ABSOLUTELY NO WARRANTY
 * This is free software, and you are welcome to redistribute it
 * under the certain conditions that can be found here
 * https://www.gnu.org/licenses/gpl-3.0.en.html
 *
 * If you have any questions or concerns, please create
 * an issue on the github page that can be found here
 * https://github.com/isXander/EvergreenHUD
 *
 * If you have a private concern, please contact
 * isXander @ business.isxander@gmail.com
 */

package dev.isxander.evergreenhud.elements.impl;

import dev.isxander.evergreenhud.elements.type.SimpleTextElement;
import dev.isxander.evergreenhud.settings.impl.BooleanSetting;
import dev.isxander.evergreenhud.elements.ElementData;
import net.minecraft.util.MathHelper;

import java.text.DecimalFormat;

public class ElementPitch extends SimpleTextElement {

    public BooleanSetting trailingZeros;

    @Override
    public void initialise() {
        addSettings(trailingZeros = new BooleanSetting("Trailing Zeros", "Display", "Add zeroes to match the accuracy.", true));
    }

    @Override
    public ElementData metadata() {
        return new ElementData("Pitch Display", "Shows the player's rotation pitch. Useful for bridging.", "Simple");
    }

    @Override
    protected String getValue() {
        if (mc.thePlayer == null)
            return "Unknown";

        return new DecimalFormat(trailingZeros.get() ? "0.0" : "#.#").format(MathHelper.wrapAngleTo180_float(mc.thePlayer.rotationPitch));
    }

    @Override
    public String getDefaultDisplayTitle() {
        return "Pitch";
    }

}

File: src/main/kotlin/dev/isxander/evergreenhud/elements/impl/ElementPitch.kt
/*
 * EvergreenHUD - A mod to improve your heads-up-display.
 * Copyright (c) isXander [2019 - 2022].
 *
 * This work is licensed under the GPL-3 License.
 * To view a copy of this license, visit https://www.gnu.org/licenses/gpl-3.0.en.html
 */

package dev.isxander.evergreenhud.elements.impl

import dev.isxander.evergreenhud.elements.type.SimpleTextElement
import dev.isxander.evergreenhud.utils.decimalFormat
import dev.isxander.evergreenhud.utils.elementmeta.ElementMeta
import dev.isxander.evergreenhud.utils.mc
import dev.isxander.settxi.impl.boolean
import dev.isxander.settxi.impl.int
import net.minecraft.util.MathHelper

@ElementMeta(id = "evergreenhud:pitch", name = "Pitch", description = "Displays the player's pitch.", category = "Player")
class ElementPitch : SimpleTextElement("Pitch", 0) {
    var accuracy by int(2) {
        range = 0..8
        name = "Accuracy"
        category = "Speed"
        description = "How many decimal places to show the speed to."
    }

    var trailingZeros by boolean(true) {
        name = "Trailing Zeros"
        category = "Speed"
        description = "Show zeros to match the accuracy."
    }

    override fun calculateValue(): String {
        return decimalFormat(accuracy, trailingZeros).format(MathHelper.wrapAngleTo180_float(mc.thePlayer?.rotationPitch ?: 0f))
    }
}
